java/java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertToVT.java

```diff
@@ -0,0 +1,132 @@
+import com.sun.source.tree.ExpressionStatementTree;
+import com.sun.source.tree.IdentifierTree;
+import com.sun.source.tree.MemberSelectTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.VariableTree;
+import com.sun.source.util.TreePath;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import org.netbeans.api.java.queries.CompilerOptionsQuery;
+import org.netbeans.api.java.source.CompilationInfo;
+import org.netbeans.api.java.source.WorkingCopy;
+import org.netbeans.modules.java.hints.errors.Utilities;
+import org.netbeans.spi.editor.hints.ErrorDescription;
+import org.netbeans.spi.editor.hints.Fix;
+import org.netbeans.spi.java.hints.ErrorDescriptionFactory;
+import org.netbeans.spi.java.hints.Hint;
+import org.netbeans.spi.java.hints.HintContext;
+import org.netbeans.spi.java.hints.JavaFix;
+import org.netbeans.spi.java.hints.TriggerPattern;
+import org.netbeans.spi.java.hints.TriggerPatterns;
+import org.openide.util.NbBundle;
+
+/**
+ *
+ * @author mjayan
+ */
+@NbBundle.Messages({
+    "DN_ConvertToVT=Convert to Virtual Thread Executor",
+    "DESC_ConvertToVT=Convert to Virtual Thread Executor",
+    "ERR_ConvertToVT=Convert to Virtual Thread Executor",
+    "FIX_ConvertToVT=Convert to Virtual Thread Executor",
+    "ERR_ConvertToThreadPerTask=Convert to newThreadPerTask executor"
+})
+@Hint(displayName = "#DN_ConvertToVT", description = "#DESC_ConvertToVT", category = "rules15",
+        minSourceVersion = "19")
+public class ConvertToVT {
+
+    private static final int VT_PREVIEW_JDK_VERSION = 19;
+
+    @TriggerPatterns({
+        @TriggerPattern(value = "$var = $expr2.$meth1($var1$);"),
+        @TriggerPattern(value = "$type $var = $expr2.$meth1($var1$);")
+    })
+    public static ErrorDescription compute(HintContext ctx) {
+        if (Utilities.isJDKVersionLower(VT_PREVIEW_JDK_VERSION) && !CompilerOptionsQuery.getOptions(ctx.getInfo().getFileObject()).getArguments().contains("--enable-preview")) {
+            return null;
+        }
+        List<String> methList = Arrays.asList("newFixedThreadPool", "newCachedThreadPool");
+        String expr = ctx.getInfo().getTrees().getElement(ctx.getVariables().get("$expr2")).getSimpleName().toString();
+        String method = ctx.getVariableNames().get("$meth1");
+        Collection<? extends TreePath> treePaths = ctx.getMultiVariables().get("$var1$");
+        boolean factory = false;
+        for (TreePath treePath : treePaths) {
+            if (treePath.getLeaf() instanceof IdentifierTree) {
+                factory = true;
+            }
+        }
+        if (expr.equals("Executors") && methList.contains(method)) {
+            Fix fix = new FixImpl(ctx.getInfo(), ctx.getPath(), factory).toEditorFix();
+            if (factory) {
+                return ErrorDescriptionFactory.forName(ctx, ctx.getPath(), Bundle.ERR_ConvertToThreadPerTask(), fix);
```

Comment:
As far as I can tell, the fix is not doing anything for `factory==true`, so maybe it should not be used at all? It is OK to have a warning if we don't/can't have a fix, but showing a fix to the user that then doing nothing feels confusing.